### ISSUE TITLE
Does not fetch signatures when installed clientside on Quilt

### ISSUE THREAD

**Issue by NoSadBeHappy** [Aizistral-Studios/No-Chat-Reports#423]
### Modloader

Quilt

### Minecraft Version

1.20.1

### Modloader Version

0.19.2

### No Chat Reports Version

1.20.1-v2.2.2

### Modpack Info

https://modrinth.com/modpack/sodiumplus

### The latest.log File

https://gist.github.com/NoSadBeHappy/43ea1b31f6fe58ce50e9578158ad7eab

### Bug Description

When chatting in my server, the server kicks me even if I have it set to allow chat reports.

### Steps to Reproduce

1. join any server
2. set no chat reports to always, meaning it allows chat reporting
3. say something in chat
4. get kicked

### Other Information

- enforce-secure-profile is set to false on the server.
- the server also has liberated chat plugin installed, but the same issue occures without the plugin.
- ![image](https://github.com/Aizistral-Studios/No-Chat-Reports/assets/95050434/2adab22a-4eb3-4331-93dc-bc40804094e8)
- Here are all of the plugins on the server.
- I can chat without no chat reports but with all other mods.
- My server is a purper server if that helps.
- ideally, I also would like to allow people to use no chat reports, and not get kicked for sending messages without chat reports as well.

**Comment by Madis0** [comment]
Set your server up properly. https://github.com/Aizistral-Studios/No-Chat-Reports/wiki/Protecting-Server-Players

**Comment by NoSadBeHappy** [comment]
I have, I installed liberated chat.

**Comment by Madis0** [comment]
But did you change the server properties as needed?

**Comment by NoSadBeHappy** [comment]
Yes. I said so in the og issue

**Comment by Madis0** [comment]
Then I'd assume it is caused by EssentialsXChat, try disabling that.

**Comment by Aizistral** [comment]
> Then I'd assume it is caused by EssentialsXChat, try disabling that.

Let me reassure you that I am not unwilling to fix plugin compatibility issues, and in fact, [I have done so in the past](https://github.com/Aizistral-Studios/No-Chat-Reports/issues/376). However, I require a conclusive report that specifies *exactly* which plugin causes the issue and under what circurmstances, so that I may reproduce it myself and figure out what action would be appropriate on our side regarding this, if any.
@NoSadBeHappy, I ask you to isolate the issue to the extent of your ability and specify every single relevant detail pertaining to it, so that I may aid in its long-term resolution.

**Comment by Aizistral** [comment]
Sorry for accidental closure, misclicked.

**Comment by NoSadBeHappy** [comment]
I don't think this is because of essentialsx chat. I have tried on a different server, and I have the same problem. As you can see in my [server.properties](https://gist.github.com/NoSadBeHappy/f133b4df58cdc5cde0e0d11f85f785ab), enforce-secure-profile is off, and here are my plugins, 
![image](https://github.com/Aizistral-Studios/No-Chat-Reports/assets/95050434/6ee3de33-dcf8-41cb-a751-e2b7f7e66584)
and I still have the same issue. When adding liberated-chat, I still have the same issue no matter what mode no-chat-reports is on. Due to this, I believe that my issue is most likely due to other mods I have, so I have sent all of my config, logs, crash-reports, and mods folders inside my .minecraft folder below. This is mostly sodium plus, as I said in the og issue, but I have added a couple of mods to it. (yes im aware I should have sensored my ip adress.) I am happy to supply any extra information to help you get this resolved, and thank you very much for your help.

.minecraft.zip
https://drive.google.com/file/d/1AL3oglOEkmBa0nNwUi4LgUV_9NjJsspW/view?usp=sharing

**Comment by Madis0** [comment]
Your clientside mods barely matter. Once you have NCR, the signing is already avoided and it is the _server_ that kicks you. 
That's why I still think it has something to do with server configuration. The plugins you isolated yourself to seem irrelevant, but Purpur itself [has a lot of options](https://purpurmc.org/docs/Configuration/), try looking over those that related to chat or signing/signatures.

**Comment by IlyaTheSawmill** [comment]
I think you should replace LiberatedChat with FreedomChat. My server also uses EssentialsXChat (and FreedomChat), but this issue did not occur. In addition, I am using the Forge 1.20.1 client and the server is Leaves.
And, why you use Minecraft 1.19.1 with NCR 1.20.1?

**Comment by NoSadBeHappy** [comment]
The reason I think it is my client is because it kicks even when I allow no chat reports to send reports. I will try freedomchat though.

**Comment by NoSadBeHappy** [comment]
Also, @IlyaIvanovsky I am using Minecraft 1.20.1, you can see in the log.

**Comment by IlyaTheSawmill** [comment]
> Also, @IlyaIvanovsky I am using Minecraft 1.20.1, you can see in the log.

Please edit your issue.
"Minecraft Version" is wrong.

**Comment by NoSadBeHappy** [comment]
> > Also, @IlyaIvanovsky I am using Minecraft 1.20.1, you can see in the log.
> 
> Please edit your issue. "Minecraft Version" is wrong.

Sorry, your right. Edited now.

**Comment by RedstoneWizard08** [comment]
> I think you should replace LiberatedChat with FreedomChat. My server also uses EssentialsXChat (and FreedomChat), but this issue did not occur. In addition, I am using the Forge 1.20.1 client and the server is Leaves. And, why you use Minecraft 1.19.1 with NCR 1.20.1?

I don't think he's using any of those chat plugins. (Essentials will mess with NCR, and call it names, and we don't like that.)

**Comment by RedstoneWizard08** [comment]
O wait he is, @NoSadBeHappy why are you using Essentials and why does it look like our cp?

**Comment by RedstoneWizard08** [comment]
Shouldn't this work with vanilla servers, though? It used to auto-disable, and this is an issue with Sodium Plus (the client), not the server. The server config shouldn't matter. Has this been tested on vanilla servers? The same thing happened to me.

**Comment by Madis0** [comment]
By the symptoms, this may be an issue with the way NCR interacts with the Quilt _client_, so the signatures may not be fetched in the first place. Aizistral has said he'll have a look at it.

**Comment by jan-Sanku** [comment]
I'm having this issue as well.

> By the symptoms, this may be an issue with the way NCR interacts with the Quilt _client_, so the signatures may not be fetched in the first place. Aizistral has said he'll have a look at it.

Indeed, I was kicked off vanilla servers with `enforce-secure-profile=false` using NCR on Quilt, regardless of NCR settings. Switched to Fabric and it works even with signatures forced on. I hope that's useful in some way.

**Comment by CompeyDev** [comment]
Facing the same issue on my client while using the Quilt loader all the time. Can confirm.

**Comment by CompeyDev** [comment]
Any updates or potential fixes? CC @Madis0 @Aizistral

**Comment by NoSadBeHappy** [comment]
Unfortunately as far as I know there isn't.

**Comment by CompeyDev** [comment]
I was speaking on the technical side of things, whether a potential fix has been identified in the mod itself; not on the users' side!

**Comment by RedstoneWizard08** [comment]
For now I'm moving Sodium Plus over to Fabric from Quilt, but this issue still needs to be fixed.

**Comment by jan-Sanku** [comment]
I feel like new versions of NCR should not be tagged as compatible with Quilt on Modrinth until this is fixed, because it's currently not really usable on Quilt.

**Comment by NoSadBeHappy** [comment]
I agree @LordVertice

**Comment by Madis0** [comment]
Per request, 1.20+ versions no longer have the Quilt marking. If you can reproduce it in earlier versions, let us know.

**Comment by CompeyDev** [comment]
Temporarily, that works, but it would be better to have this issue sorted out in the future. Been using fabric for a couple months instead of quilt which isn't ideal, since there's some mods I simply cannot use.

**Comment by jan-Sanku** [comment]
any updates on this one? i see this has the "3rd party" label now, is there anyone in particular this needs to be reported to?
